src/main/scala/org/allenai/plugins/VersionInjectorPlugin.scala

```diff
@@ -72,6 +91,14 @@ object VersionInjectorPlugin extends AutoPlugin {
     (gitCommand("describe").!!).trim
   }
 
+  def gitMostRecentCommit(path: File) = {
+    gitCommand("log", "-1", "--format=%H", path).!!.trim
+  }
+
+  val gitMRCTask = gitMRC := {
+    gitMostRecentCommit(sourceDirectory.value)
```

Comment:
This should use the project base (`baseDirectory.value`). It's possible that some other files (like `build.sbt`) will affect the output jar - and the number of extra misses due to this will be fairly small.